pkg/apis/core/v1alpha1/types.go

```diff
@@ -310,13 +310,15 @@ type CDIConfigSpec struct {
 	UploadProxyURLOverride   *string                      `json:"uploadProxyURLOverride,omitempty"`
 	ScratchSpaceStorageClass *string                      `json:"scratchSpaceStorageClass,omitempty"`
 	PodResourceRequirements  *corev1.ResourceRequirements `json:"podResourceRequirements,omitempty"`
+	FeatureGates             []string                     `json:"featureGates,omitempty"`
 }
 
 //CDIConfigStatus provides
 type CDIConfigStatus struct {
 	UploadProxyURL                 *string                      `json:"uploadProxyURL,omitempty"`
 	ScratchSpaceStorageClass       string                       `json:"scratchSpaceStorageClass,omitempty"`
 	DefaultPodResourceRequirements *corev1.ResourceRequirements `json:"defaultPodResourceRequirements,omitempty"`
+	FeatureGates                   []string                     `json:"featureGates,omitempty"`
```

Comment:
When would be this be different from spec?  Are there default feature gates?  Where are they stored?

Reply:
Similarly are there times when we won't all certain feature gates to be explicitly set?

Just trying to understand both properties

Reply:
The way I understand this, if the value is set, then the feature gate is on, if it is not set, it is off

Reply:
I did it by analogy to other config variables in CDI, so the user sets this on Spec, and the results are available on Status after it is reconciled and defaults are set. But maybe we do not need any defaults at all. 

Ideally we would do something like this:
1. We create a new feature that has to be off by default (backward compatibility or something), and it is set to ON by using feature gates, the logic checks to find feature gate. 
2. In the future - we decide all backward incompatible clients/users are not longer available so now we make the feature ON by default. To do this we remove the logic that checks feature gate, and remove feature gate. 

Is it ok? If not, then we need to set defaults somewhere, and in my opinion we can handle them in CDIConfig reconciliation  loop (then - the final value is nicely visible to the user i CDIConfig.Status), or maybe we should just handle defaults in FeatureGates?

Reply:
It's still not clear to me when `spec. featureGates != status.featureGates`.  Can you give an example?  If not, then I say get rid of the status.

Reply:
That should teach me to look closer, yes I don't think we need the status if the spec is always the same as the spec.

Reply:
good, no need for additional complexity